bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/models/AbstractScalarResponse.java

```diff
@@ -0,0 +1,175 @@
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractScalarResponse {
+    /**
+     * An abstract method to allow the caller to get the payload of the response
+     * 
+     * @return a non-null json array
+     */
+    protected abstract @Nullable JsonArray getPayload();
+
+    /**
+     * Converts this generic response into the specified type
+     *
+     * @param <T> the generic type that will be returned
+     * @param clazz the class to cast to
+     * @return the object cast to class
+     * @throws IOException Signals that an I/O exception has occurred.
+     */
+    public <T> T as(final Class<T> clazz) throws IOException {
+        Objects.requireNonNull(clazz, "clazz cannot be null");
+
+        // First see if there is a constructor that takes a ScalarWebResult (us)
+        // If so - call it with us
+        // Otherwise try to use GSON to construct the class and set the fields
+        try {
+            final Constructor<T> constr = clazz.getConstructor(this.getClass());
+            return constr.newInstance(this);
+        } catch (final NoSuchMethodException e) {
+            final JsonArray localResults = getPayload();
+            if (localResults == null || isBlank(localResults)) {
+                throw new IllegalArgumentException(
+                        "Cannot convert ScalarWebResult for " + clazz + " with results: " + localResults);
+            } else if (localResults.size() == 1) {
+                JsonElement elm = localResults.get(0);
+                if (elm.isJsonArray()) {
+                    final JsonArray arry = elm.getAsJsonArray();
+                    if (arry.size() == 1) {
+                        elm = arry.get(0);
+                    } else {
+                        elm = arry;
+                    }
+                }
+                final Gson gson = GsonUtilities.getApiGson();
+
+                if (elm.isJsonObject()) {
+                    final JsonObject jobj = elm.getAsJsonObject();
+                    return gson.fromJson(jobj, clazz);
+                } else {
+                    if (SonyUtil.isPrimitive(clazz)) {
+                        return gson.fromJson(elm, clazz);
+                    } else {
+                        throw new IllegalArgumentException(
+                                "Cannot convert ScalarWebResult to " + clazz + " with results: " + localResults, e);
+                    }
+                }
+            }
+            throw new IllegalArgumentException(
+                    "Cannot convert ScalarWebResult to " + clazz + " with results: " + localResults, e);
+
+        } catch (InstantiationException | IllegalAccessException | IllegalArgumentException
+                | InvocationTargetException e) {
+            throw new IllegalArgumentException(
+                    "Cannot convert ScalarWebResult to " + clazz + " for reason: " + e.getMessage(), e);
```

Comment:
You are wrapping a IllegalArgumentException in yet another IllegalArgumentException?

Reply:
lol!